### ISSUE TITLE
Coalescing of Diet ranges

### ISSUE THREAD

**Issue by ajatix** [typelevel/cats-collections#188]
Consider this example

```scala
val d: Diet[Int] = Diet.empty
val d2 = d.addRange(2 toIncl 20) // { [2, 20] }
val d3 = d2.removeRange(5 toIncl 15) // { [2, 4], [16, 20] }
val d4 = d3.addRange(5 toIncl 15) // { [2, 15], [16, 20] } <-- no longer referentially transparent
```

I would have expected `d4` to be `{ [2, 20] }`

**Comment by larsrh** [comment]
What version is that? I can't reproduce it on latest `master`.

**Comment by ajatix** [comment]
I tried this on `0.7.0`

**Comment by ajatix** [comment]
Found the fix that was added after `0.7.0` in the `noMoreThan` and `noLessThan` methods - https://github.com/typelevel/cats-collections/commit/074bf911dc1988c54c0fd1a9aed8d5fdcba5ed63#diff-57bfa0da95ba1d55a24c48b028296023

**Comment by larsrh** [comment]
Right, that may have been the same problem. Because it wasn't a regression, I didn't release a hotfix. But maybe I should.
